From 0dcac3519396df51d5f4133574786bf3b0089140 Mon Sep 17 00:00:00 2001
From: Huy Bui <huy.bui.wm@renesas.com>
Date: Wed, 7 May 2025 00:13:13 +0700
Subject: [PATCH 28/52] arm64: dts: renesas: ironhide: Add I2C0, I2C1 and
 EEPROM

Enable the I2C0, I2C1 bus on the Ironhide board, and describe the I2C
EEPROM present.

Signed-off-by: Minh Le <minh.le.aj@renesas.com>
Signed-off-by: Hai Pham <hai.pham.ud@renesas.com>
---
 arch/arm/dts/r8a78000-ironhide.dts | 41 ++++++++++++++++++++++++++++++
 1 file changed, 41 insertions(+)

diff --git a/arch/arm/dts/r8a78000-ironhide.dts b/arch/arm/dts/r8a78000-ironhide.dts
index 43c5aa308d5..f17a93085cf 100644
--- a/arch/arm/dts/r8a78000-ironhide.dts
+++ b/arch/arm/dts/r8a78000-ironhide.dts
@@ -13,6 +13,15 @@
 	compatible = "renesas,ironhide", "renesas,r8a78000";
 
 	aliases {
+		i2c0 = &i2c0;
+		i2c1 = &i2c1;
+		i2c2 = &i2c2;
+		i2c3 = &i2c3;
+		i2c4 = &i2c4;
+		i2c5 = &i2c5;
+		i2c6 = &i2c6;
+		i2c7 = &i2c7;
+		i2c8 = &i2c8;
 		serial0 = &hscif0;
 	};
 
@@ -83,6 +92,28 @@
 	status = "okay";
 };
 
+&i2c0 {
+	pinctrl-0 = <&i2c0_pins>;
+	pinctrl-names = "default";
+
+	status = "okay";
+	clock-frequency = <400000>;
+
+	eeprom@50 {
+		compatible = "rohm,br24g01", "atmel,24c01";
+		reg = <0x50>;
+		pagesize = <8>;
+	};
+};
+
+&i2c1 {
+	pinctrl-0 = <&i2c1_pins>;
+	pinctrl-names = "default";
+
+	status = "okay";
+	clock-frequency = <400000>;
+};
+
 &pfc {
 	pinctrl-0 = <&scif_clk_pins>;
 	pinctrl-names = "default";
@@ -92,6 +123,16 @@
 		function = "hscif0";
 	};
 
+	i2c0_pins: i2c0 {
+		groups = "i2c0";
+		function = "i2c0";
+	};
+
+	i2c1_pins: i2c1 {
+		groups = "i2c1";
+		function = "i2c1";
+	};
+
 	scif_clk_pins: scif-clk {
 		groups = "scif_clk";
 		function = "scif_clk";
-- 
2.34.1

